pkg/diagnostics/context.go

```diff
@@ -85,6 +87,50 @@ func dedupeWarnings(warnings []string) []string {
 	return uniqueWarnings
 }
 
+func getContextTfeRunInfo(ctx context.Context) (*TfeRunInfo, error) {
+	contextTfeRunInfo := ctx.Value(tfeRunInfoContextKey)
+	if contextTfeRunInfo != nil {
+		return contextTfeRunInfo.(*TfeRunInfo), nil
+	} else {
+		return nil, errors.New("Context does not have TFE run info")
+	}
```

Comment:
```suggestion
	if contextTfeRunInfo == nil {
	    return nil, errors.New("Context does not have TFE run info")
	} 
	return contextTfeRunInfo.(*TfeRunInfo), nil
```
These express the same idea, But I like trying to structure it this way so that we deal with errors first and "short circuit" the code